actionpack/lib/action_dispatch/middleware/server_timing.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "active_support/notifications"
+
+module ActionDispatch
+  class ServerTiming
+    SERVER_TIMING_HEADER = "Server-Timing"
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      events = []
+      subscriber = ActiveSupport::Notifications.subscribe(/.*/) do |*args|
+        events << ActiveSupport::Notifications::Event.new(*args)
+      end
+
+      status, headers, body = @app.call(env)
+      ActiveSupport::Notifications.unsubscribe(subscriber)
+
+      header_info = []
+      events.group_by(&:name).each do |event_name, events_collection|
+        header_info << "#{event_name};dur=#{duration_sum(events_collection)}"
```

Comment:
This could be `events_collection.sum(&:duration)` instead of `duration_sum(events_collection)`.

Reply:
Nice suggestions, done in 45c9631614437b32af264c3fea3dad0333222456.